source/UOXJSMethods.cpp

```diff
@@ -3642,36 +3650,44 @@ JSBool CBase_UseResource( JSContext *cx, JSObject *obj, uintN argc, jsval *argv,
 
 	UI32 amount		= static_cast<UI32>( JSVAL_TO_INT( argv[0] ));
 	UI16 realId		= static_cast<UI16>( JSVAL_TO_INT( argv[1] ));
-	UI16 itemColour = 0;
-	UI32 moreVal	= 0;
+	SI32 itemColour = 0;
+	SI64 moreVal	= -1;
+	std::string sectionId = "";
 
-	// Min. 2 Arguments (amount + id) or max 4 (amount + id + color + moreVal)
-	if(( argc < 2 ) || ( argc > 4 ))
+	// Min. 2 Arguments (amount + id) or max 5 (amount + id + color + moreVal + sectionId)
+	if(( argc < 2 ) || ( argc > 5 ))
 	{
-		MethodError( "(UseResource) Invalid count of parameters: %d, either needs 2, 3 or 4", argc );
+		MethodError( "(UseResource) Invalid count of parameters: %d, needs from 2 to 5 parameters", argc );
 		return JS_FALSE;
 	}
 
 	if( argc >= 3 )
 	{
-		itemColour = static_cast<UI16>( JSVAL_TO_INT( argv[2] ));
+		itemColour = static_cast<SI32>( JSVAL_TO_INT( argv[2] ));
 	}
 	if( argc >= 4 )
 	{
-		moreVal = static_cast<UI16>( JSVAL_TO_INT( argv[3] ));
+		moreVal = static_cast<SI64>( JSVAL_TO_INT( argv[3] ));
```

Comment:
```suggestion
		moreVal = static_cast<SERIAL>( JSVAL_TO_INT( argv[3] ));
```
Though not sure if you're calling this with -1?  If so, that's probably why the larger one ...